src/problems.jl

```diff
@@ -40,7 +40,7 @@ end
 
 # constructor if model is not specified
 function Problem(head::Symbol, objective::AbstractExpr, constraints::Array=Constraint[],
-                 solver::MathProgBase.AbstractMathProgSolver=get_default_solver())
+                 solver::MathProgBase.AbstractMathProgSolver=DEFAULT_SOLVER)
```

Comment:
We probably should introduce a function

```
check_default_solver(::Any)=nothing
check_default_solver(::Nothing)=`error("DEFAULT_SOLVER not set, you can ...."`
```

since right now the helpful error message is in `get_default_solver`,
which is no longer called.

Reply:
That's true, but I think that's the desired behaviour to avoid spam. Currently, `get_default_solver` will be run on package load anyways.

https://github.com/JuliaOpt/Convex.jl/blob/04afdf5b23cfcd77c6f28f74457594df850f160e/src/solver_info.jl#L51

Reply:
Shouldn't this be in `__init__`  as else it will not work with precompilation?

(the whole block https://github.com/JuliaOpt/Convex.jl/blob/04afdf5b23cfcd77c6f28f74457594df850f160e/src/solver_info.jl#L43-L60)

Reply:
That's true. If folks want we could just move that code into an `__init__`, which would be a less breaking change. At this point, I'd rather just deprecate this interface anyways.

Reply:
Actually, I guess we could do both. I could move the code into an `__init__` (probably a separate PR?) for an earlier release, but still deprecate the default solver functionality. This would help move the interface closer to JuMP, but allow us to provide a longer deprecation cycle.